### PR TITLE
chore(Refactor): Reorganize RulesTable component

### PR DESCRIPTION
**Pull request by adonispuente**
Learning from my last large attempt at refactoring, This PR reorganizes just the RulesTable component. There should be no changes in functionality. This simply cleans up the RulesTable component so that its much easier to navigate. The file was 800+ lines of code and it caused of lot of jumping around when trying to debug the table. This PR cuts it in half and utilizes the helpers.js file

# Checklist:

- [ ] The commit message has the Jira ticket linked
- [x] PR has a short description
- [ ] Screenshots before and after the change are added
- [ ] Tests for the changes have been added
- [ ] README.md is updated if necessary
- [ ] Needs additional dependent work

### CODE REVIEW

File: src/PresentationalComponents/RulesTable/RulesTable.js [hunk 1]
```diff
@@ -173,60 +153,12 @@ const RulesTable = ({ isTabActive }) => {
     setFilters({ ...filters, offset: newOffset });
   };
 
-  const toggleRulesDisabled = (rule_status) => {
```
Comment (mkholjuraev) [inline src/PresentationalComponents/RulesTable/RulesTable.js:176]: I would vote for leaving functions, like this, that are specific to a component inside. This function deals with updating a state only. Functions that can be reused in other components are great to be refactored

File: src/PresentationalComponents/RulesTable/RulesTable.js [hunk 2]
```diff
@@ -284,161 +234,13 @@ const RulesTable = ({ isTabActive }) => {
       if (rules.data.length === 0) {
         setRows(emptyRows(filters, toggleRulesDisabled));
       } else {
-        const rows = rules.data.flatMap((value, key) => [
```
Comment (mkholjuraev) [inline src/PresentationalComponents/RulesTable/RulesTable.js:287]: Happy to see this got refactored!

**Review by mkholjuraev — COMMENTED** [review]
Great for initial refactoring. I believe that in the next phase, we can try to see how those extracted functions could be minimized.

File: src/PresentationalComponents/RulesTable/RulesTable.js (hunk 1 above)
Comment (adonispuente) [reply src/PresentationalComponents/RulesTable/RulesTable.js:176]: This makes sense, I've moved it back and just passed it into the filter config for the onChange function

**Pull request merged**